core/src/com/unciv/logic/map/mapgenerator/MapLandmassGenerator.kt

```diff
@@ -152,7 +152,10 @@ class MapLandmassGenerator(val ruleset: Ruleset, val randomness: MapGenerationRa
     }
 
     private fun createPangaea(tileMap: TileMap) {
-        do {
+        val largeContinentThreshold = (tileMap.values.size / 4).coerceAtMost(25)
+        var maxRetries = 200  // A bit much but enough to reach waterThreshold -1
+
+        while(--maxRetries >= 0) {
```

Comment:
Don't use a parameter called "max retries" to save the number of retries left :(